test/gst-msdk/decode/10bit/av1.py

```diff
@@ -23,7 +23,7 @@ def before(self):
   def test(self, case):
     vars(self).update(spec[case].copy())
 
-    dxmap = {".ivf" : "ivfparse", ".webm" : "matroskademux"}
+    dxmap = {".ivf" : "ivfparse", ".webm" : "matroskademux", ".obu" : "av1parse"}
```

Comment:
Hmm... we already add `av1parse` unconditionally, below, in `gstdecoder`.  This will produce `gstdecoder = "av1parse ! av1parse ! msdkav1dec"`.  We don't want this.

How about something like:
```python
dxmap = {".ivf" : "ivfparse", ".webm" : "matroskademux", ".obu" : "av1parse"}
ext = os.path.splitext(self.source)[1]
dx = dxmap.get(ext, None)
assert dx is not None, "Unrecognized source file extension {}".format(ext)

if "av1parse" not in dx:
  dx += " ! av1parse"

vars(self).update(
  case             = case,
  gstdecoder  = "{} ! msdkav1dec".format(dx),
)
```

...similar should be done for gst-vaapi/decode/10bit/av1.py

Reply:
updated, thanks